src/workerd/api/global-scope.c++

```diff
@@ -396,14 +396,20 @@ kj::Promise<WorkerInterface::AlarmResult> ServiceWorkerGlobalScope::runAlarm(
       // completes we want to cancel the alarm handler. If the alarm handler promise completes first
       // timeout will be canceled.
       auto timeoutPromise = context.afterLimitTimeout(timeout).then([&context]() -> kj::Promise<WorkerInterface::AlarmResult> {
-        LOG_NOSENTRY(WARNING, "Alarm exceeded its allowed execution time");
         // We don't want to delete the alarm since we have not successfully completed the alarm
         // execution.
         auto& actor = KJ_ASSERT_NONNULL(context.getActor());
         auto& persistent = KJ_ASSERT_NONNULL(actor.getPersistent());
         persistent.cancelDeferredAlarmDeletion();
+
+        LOG_NOSENTRY(WARNING, "Alarm exceeded its allowed execution time");
+        // Report alarm handler failure and log it.
+        auto e = KJ_EXCEPTION(OVERLOADED, "broken.dropped; worker_do_not_log; Alarm exceeded its allowed execution time");
```

Comment:
@justin-mp updated the PR to include jsg logging, but I'm not going to use the `JSG_KJ_EXCEPTION` macro. Instead, I did as we do here: https://github.com/cloudflare/workerd/blob/9ea0e090d7f32e89239a6c687a25ef70376ea906/src/workerd/api/actor-state.c%2B%2B#L817-L836
manually appending the `jsg.Error: ` prefix. 
If I used the macro this would be appended in the beginning, and we don't filter this out properly later on.